code/modules/client/preferences.dm

```diff
@@ -1338,6 +1369,22 @@ var/list/preferences_datums = list()
 			switch(href_list["preference"])
 
 				//citadel code
+				if("genital_colour")
+					switch(features["genitals_use_skintone"])
+						if(TRUE)
+							features["genitals_use_skintone"] = FALSE
+						if(FALSE)
+							features["genitals_use_skintone"] = TRUE
+							features["cock_color"] = sanitize_hexcolor(skintone2hex(skin_tone))
+							features["breasts_color"] = sanitize_hexcolor(skintone2hex(skin_tone))
+							features["vag_color"] = sanitize_hexcolor(skintone2hex(skin_tone))
+							features["balls_color"] = sanitize_hexcolor(skintone2hex(skin_tone))
```

Comment:
These are in place so that the preferences page updates the selected colours shown. If you toggle the skintone option, it will now change the colour on the right to match the skintone.
In hindsight, the s_tone at 1182 needs an if-statement so it only updates if the option is enabled, like I did for line 861.